src/vs/workbench/contrib/positronPlots/browser/positronPlotsService.ts

```diff
@@ -164,6 +183,34 @@ export class PositronPlotsService extends Disposable implements IPositronPlotsSe
 			}
 		});
 
+		// When the extension service is about to stop, remove any HTML plots
+		// from the plots list. These plots are backed by a proxy that runs in
+		// the extension host, so may become invalid when the extension host is
+		// stopped.
+		this._register(this._extensionService.onWillStop((e) => {
+			// Nothing to do if there are no plots
+			if (this._plots.length === 0) {
+				return;
+			}
+			let removedSelectedPlot = false;
+			this._plots.forEach((plot) => {
```

Comment:
This can be:

```JavaScript
this._plots.forEach((plot, index) => {
```

Which will remove the need to call:

```JavaScript
    this._plots.indexOf(plot)
```

On line 202.